embedded/sql/engine.go

```diff
@@ -1267,16 +1267,25 @@ func (e *Engine) QueryPreparedStmt(stmt DataSource, params map[string]interface{
 	// TODO: eval params at once
 	nparams, err := normalizeParams(params)
 	if err != nil {
+		if tx == nil {
```

Comment:
what about deferring the close and checking if there was an error... the benefit is to avoid having a close for each error handling...

```
defer func() {
    if err != nil {
        qtx.Cancel()
    }
}```